src/ship.js

```diff
@@ -0,0 +1,34 @@
+class Ship {
+    constructor(itinerary) {
+        this.itinerary = itinerary;
+        this.currentPort = itinerary.ports[0];
+        this.previousPort = null;
+        this.currentPort.addShip(this);
+    }
+
+    setSail() {
+        this.currentPort.removeShip(this);
```

Comment:
For better formatting I would move this down, between line19 this.previousPort and line 20 this.currentPort